Sming/Core/Data/Stream/MemoryDataStream.cpp

```diff
@@ -82,3 +90,23 @@ int MemoryDataStream::seekFrom(int offset, unsigned origin)
 	readPos = newPos;
 	return readPos;
 }
+
+String MemoryDataStream::readString(size_t maxLen)
+{
+	// Are we're after the entire contents of this stream?
+	if(readPos != 0 || maxLen < size) {
+		// No: do a regular read
+		return IDataSourceStream::readString(maxLen);
+	}
+
+	// Move content into the String object
+	ensureCapacity(size + 1); // Ensure size < capacity
+	String s;
+	if(s.setBuffer({buffer, capacity, size})) {
+		buffer = nullptr;
+		readPos = 0;
+		size = 0;
+		capacity = 0;
+	}
+	return s;
```

Comment:
I'm on the fence about this behaviour. It's probably correct in 99% of situations, but the safer option might be to add an overload such as `size_t readString(String& s);`

Reply:
> but the safer option might be to add an overload such as size_t readString(String& s);

Got for it as long as the safer option does not impact compatibility, memory usage and speed.